### ISSUE TITLE
[feature request] Netatmo 'boiler_status' binary switch

### ISSUE THREAD

**Issue by lboue** [domoticz/domoticz#3039]
Hello,

Could you add in the Netatmo plugin a new binary switch synchronised with the `boiler_status `variable?
I would be able to be notified when boiler is turning on/off..

Here is my getHomeStatus json result (check `type': 'NATherm1'`)
```
{   'body': {   'home': {   'id': '**************************',
                            'modules': [   {   'firmware_revision': 197,
                                               'id': '70:ee:50:xx:xx:xx',
                                               'rf_strength': 107,
                                               'type': 'NAPlug',
                                               'wifi_strength': 26},
                                           {   'anticipating': False,
                                               'battery_level': 3753,
                                               'battery_state': 'high',
                                               'boiler_status': False,
                                               'boiler_valve_comfort_boost': False,
                                               'bridge': '70:ee:50:ab:cd:ef',
                                               'firmware_revision': 71,
                                               'id': '04:00:00:xx:xx:xx',
                                               'reachable': True,
                                               'rf_strength': 39,
                                               'type': 'NATherm1'},
                                           {   'battery_level': 3213,
                                               'battery_state': 'full',
                                               'bridge': '70:ee:50:ab:cd:aa',
                                               'firmware_revision': 73,
                                               'id': '09:00:00:xx:xx:xx',
                                               'reachable': True,
                                               'rf_strength': 82,
                                               'type': 'NRV'},
                                           {   'battery_level': 3081,
                                               'battery_state': 'full',
                                               'bridge': '70:ee:50:ab:cd:bb',
                                               'firmware_revision': 73,
                                               'id': '09:00:00:xx:xx:xx',
                                               'reachable': True,
                                               'rf_strength': 55,
                                               'type': 'NRV'}],
```

Regards,

**Comment by lboue** [comment]
It should be something like it is for `therm_setpoint_mode `to test `boiler_status` variable here: https://github.com/domoticz/domoticz/blob/80fcf90238d76a81b85ad1e077d8ac6cac935a94/hardware/Netatmo.cpp#L1494

Here is proposal at the end:

```
				m_thermostatDeviceID[roomID & 0xFFFFFF] = id;
				m_thermostatModuleID[roomID & 0xFFFFFF] = id;

				if (!room["therm_measured_temperature"].empty())
				{
					SendTempSensor(roomID, 255, room["therm_measured_temperature"].asFloat(), roomName);
				}
				if (!room["therm_setpoint_temperature"].empty())
				{
					SendSetPointSensor((roomID & 0XFF0000) >> 16, (roomID & 0XFF00) >> 8, roomID & 0XFF, room["therm_setpoint_temperature"].asFloat(), roomName);
				}
				if (!room["therm_setpoint_mode"].empty())
				{
					std::string setpoint_mode = room["therm_setpoint_mode"].asString();
					bool bIsAway = (setpoint_mode == "away");
					std::string aName = "Away " + roomName;
					SendSwitch(roomID & 0xFFFFFF, 1, 255, bIsAway, 0, aName);
				}
				if (!room["boiler_status"].empty())
				{
					std::string boiler_status = room["boiler_status"].asString();
					bool bBoilerOn = (boiler_status == "True");
					std::string aName = "Boiler status";
					SendSwitch(roomID & 0xFFFFFF, 1, 255, bBoilerOn, 0, aName);
				}
```

I can test it if someone push it on the dev branch.

Regards,

**Comment by gizmocuz** [comment]
@lboue , it would be great if you could make the code changes as they look good!
If you're on Windows, please download the visual studio 2017 community edition (free)
for more instructions see:
https://github.com/domoticz/domoticz/blob/development/INSTALL.md

You can also contact me on the forum if you want me to implement it, but i need some (temporarily) credentials.

But if you can compile yourself, why not add the code block and test ?

**Comment by lboue** [comment]
I will try to compile code myself. I have never compiled domoticz before. 
Thanks for your avdvice. I'll let you know in the coming days.

**Comment by gizmocuz** [comment]
Ahhh okey, because you said "push it on the dev branch"
If you are using a raspberry or ubuntu/debian, please follow:
https://www.domoticz.com/wiki/Raspberry_Pi_-_Build_Domoticz_from_source

**Comment by lboue** [comment]
- Installing Visual studio 2017 community edition --> OK
- Unpacking the Windows libs archive inside the "msbuild" folder --> OK
- Building project --> OK

But when I run the deamon and browse the home page I got a HTTP 404 page.

![image](https://user-images.githubusercontent.com/938089/52747274-b149db80-2fe3-11e9-9ddf-6793bcd0b0ce.png)

**Comment by gizmocuz** [comment]
You have to set the working folder in visual studio.
Right mouse on the project (in the solution explorer), properties, Configuration Properties->Debugging
Set "Working Directory" to "$(ProjectDir)/.."

**Comment by gizmocuz** [comment]
(If this is not mentioned in the INSTALL.TXT file.. make you could make a PR for this ;)

**Comment by lboue** [comment]
I set working folder in visual studio as following and build binary again:

![image](https://user-images.githubusercontent.com/938089/52805049-03433d80-3086-11e9-99cf-dd49f3f7c348.png)

But I am still getting the same error:
```
2019-02-14 18:34:14.572  Error: Webserver: File '/index.html': No such file or directory (2)  (remote address: 127.0.0.1)
2019-02-14 18:34:14.787  Error: Webserver: File '/favicon.ico': No such file or directory (2)  (remote address: 127.0.0.1)
```
Could you share your _domoticz.vcxproj.user_ file or upload it to github?
Web files seems to be here : $(ProjectDir)/../www

Regards,

**Comment by gizmocuz** [comment]
did you manage to solve this ? it should be $(ProjectDir)/..

**Comment by olivierberthomme** [comment]
Hi @lboue, any chance you confirm issue solved with pull request #3811 ?
Certainly you'll need to recreate the netatmo hardware to get status switch.

**Comment by waltervl** [comment]
No feedback, issue can be closed.
